### PR TITLE
Simplify  a bit

### PR DESCRIPTION
**Pull request by jespera**
Followup PR with additional fixes

### CODE REVIEW

**Comment by MikkelThomsen** [comment]
I suggest a unit test of the `valueToJson`

**Comment by jespera** [comment]
> I suggest a unit test of the `valueToJson`

Tests added.

**Comment by zoren** [comment]
Should we test if there are less than 2 elements in the array of elements and fail early?

**Comment by jespera** [comment]
I added checking in `mkTupleValue` to ensure we don't create tuples with less than two values.

File: lib/DeonData.ts [hunk 1]
```diff
@@ -276,8 +276,12 @@ export interface TupleValue {
   class: 'TupleValue';
   values: Value[];
 }
-export const mkTupleValue = (values: Value[]): TupleValue =>
-  ({ values, class: 'TupleValue' });
+export const mkTupleValue = (values: Value[]): TupleValue => {
```
[the remark this answers is not in the document]
Comment (jespera) [reply lib/DeonData.ts]: That is really really cool!

**Pull request merged**